### PR TITLE
Feature/transaction details

### PR DESCRIPTION
**Pull request by Dreyhh**
- Added logic to save TransactionMetadata (formerly Label) to the relays and to default using the saved values (if any) if no new ones are being provided by the user.
- Added support to provide BtcExchangeRates for each transaction
-   Updated signature and logic of saveLabel() method to handle multiple TransactionMetadata updates
-   Refactored methods related to augmented transactions to reuse shared logic.

### CODE REVIEW

File: src/SmartVaults.ts [hunk 1]
```diff
@@ -2056,48 +2057,58 @@ export class SmartVaults {
...
+   * @param {SmartVaultsTypes.TransactionMetadata} label - The label object containing the data to be saved.
+   * @returns {Promise<SmartVaultsTypes.PublishedTransactionMetadata>} - A promise that resolves to the published label.
    * 
    * @throws {Error} - Throws an error if the policy event retrieval fails, or if shared keys are not found.
    * 
    * @example
    * const publishedLabel = await saveLabel('some-policy-id', { data: { 'Address': 'some-address' }, text: 'some-label-text' });
    */
-  async saveLabel(policyId: string, label: SmartVaultsTypes.Label): Promise<SmartVaultsTypes.PublishedLabel> {
+  saveLabel = async (policyId: string, label: SmartVaultsTypes.TransactionMetadata | Array<SmartVaultsTypes.TransactionMetadata>): Promise<Array<SmartVaultsTypes.PublishedTransactionMetadata>> => {
```
Comment (sebastianmontero) [inline src/SmartVaults.ts:2068]: I think we should update the name of this method to be saveTransactionMetadata, also code within the method still references label in various points that should be renamed, including the event kind

File: src/SmartVaults.ts [hunk 2]
```diff
@@ -2056,48 +2057,58 @@ export class SmartVaults {
...
+
+    await Promise.all(promises)
+
+    return publishedLabels
 
-    return publishedLabel
   }
 
-  private async _getLabels(filter: Filter<SmartVaultsKind.Labels>[]): Promise<SmartVaultsTypes.PublishedLabel[]> {
+  private async _getLabels(filter: Filter<SmartVaultsKind.Labels>[]): Promise<SmartVaultsTypes.PublishedTransactionMetadata[]> {
```
Comment (sebastianmontero) [inline src/SmartVaults.ts:2111]: We should updated the name of the method

File: src/SmartVaults.ts [hunk 3]
```diff
@@ -2108,12 +2119,12 @@ export class SmartVaults {
...
    * @async
    * @param {PaginationOpts} [paginationOpts={}] - Optional pagination options for fetching labels.
-   * @returns {Promise<SmartVaultsTypes.PublishedLabel[]>} - A promise that resolves to an array of published labels.
+   * @returns {Promise<SmartVaultsTypes.PublishedTransactionMetadata[]>} - A promise that resolves to an array of published labels.
    *
    * @example
    * const labels = await getLabels();
    */
-  async getLabels(paginationOpts: PaginationOpts = {}): Promise<SmartVaultsTypes.PublishedLabel[]> {
+  async getLabels(paginationOpts: PaginationOpts = {}): Promise<SmartVaultsTypes.PublishedTransactionMetadata[]> {
```
Comment (sebastianmontero) [inline src/SmartVaults.ts:2127]: We should update the name of the method

File: src/SmartVaults.ts [hunk 4]
```diff
@@ -2128,13 +2139,13 @@ export class SmartVaults {
...
    * @param {PaginationOpts} [paginationOpts={}] - Optional pagination options.
-   * @returns {Promise<Map<string, SmartVaultsTypes.PublishedLabel | Array<SmartVaultsTypes.PublishedLabel>>>} - 
+   * @returns {Promise<Map<string, SmartVaultsTypes.PublishedTransactionMetadata | Array<SmartVaultsTypes.PublishedTransactionMetadata>>>} - 
    * A promise that resolves to a map where the keys are policy IDs and the values are the associated labels.
    *
    * @example
    * const labelsMap = await getLabelsByPolicyId(['policy1', 'policy2']);
    */
-  getLabelsByPolicyId = async (policy_ids: string[] | string, paginationOpts: PaginationOpts = {}): Promise<Map<string, SmartVaultsTypes.PublishedLabel | Array<SmartVaultsTypes.PublishedLabel>>> => {
+  getLabelsByPolicyId = async (policy_ids: string[] | string, paginationOpts: PaginationOpts = {}): Promise<Map<string, SmartVaultsTypes.PublishedTransactionMetadata | Array<SmartVaultsTypes.PublishedTransactionMetadata>>> => {
```
Comment (sebastianmontero) [inline src/SmartVaults.ts:2148]: we should update the name of the method

File: src/SmartVaults.ts [hunk 5]
```diff
@@ -2148,13 +2159,13 @@ export class SmartVaults {
...
    * @param {PaginationOpts} [paginationOpts={}] - Optional pagination options.
-   * @returns {Promise<Map<string, SmartVaultsTypes.PublishedLabel>>} - 
+   * @returns {Promise<Map<string, SmartVaultsTypes.PublishedTransactionMetadata>>} - 
    * A promise that resolves to a map where the keys are label IDs and the values are the corresponding labels.
    *
    * @example
    * const labelsMap = await getLabelById(['label1', 'label2']);
    */
-  async getLabelById(label_ids: string[] | string, paginationOpts: PaginationOpts = {}): Promise<Map<string, SmartVaultsTypes.PublishedLabel>> {
+  async getLabelById(label_ids: string[] | string, paginationOpts: PaginationOpts = {}): Promise<Map<string, SmartVaultsTypes.PublishedTransactionMetadata>> {
```
Comment (sebastianmontero) [inline src/SmartVaults.ts:2168]: we should update the name of the method

File: src/SmartVaults.ts [hunk 6]
```diff
@@ -2168,20 +2179,20 @@ export class SmartVaults {
...
-   * @returns {Promise<SmartVaultsTypes.PublishedLabel>} - 
-   * A promise that resolves to a PublishedLabel.
+   * @returns {Promise<SmartVaultsTypes.PublishedTransactionMetadata>} - 
+   * A promise that resolves to a PublishedTransactionMetadata.
    *
    * @example
    * const labels = await getLabelByLabelData("policyId","trxid");
    */
-  async getLabelByLabelData(policyId: string, labelData: string): Promise<SmartVaultsTypes.PublishedLabel> {
+  async getLabelByLabelData(policyId: string, labelData: string): Promise<SmartVaultsTypes.PublishedTransactionMetadata> {
```
Comment (sebastianmontero) [inline src/SmartVaults.ts:2188]: we should update the name of the method

File: src/event-kind-handler/PolicyHandler.ts [hunk 7]
```diff
@@ -31,7 +31,8 @@ export class PolicyHandler extends EventKindHandler {
     | Array<PublishedApprovedProposal>>>
   private readonly getSharedSigners: (publicKeys?: string | string[]) => Promise<PublishedSharedSigner[]>
   private readonly getOwnedSigners: () => Promise<PublishedOwnedSigner[]>
-  private readonly getLabelsByPolicyId: (policy_ids: string[] | string, paginationOpts: PaginationOpts) => Promise<Map<string, PublishedLabel | Array<PublishedLabel>>>
```
Comment (sebastianmontero) [inline src/event-kind-handler/PolicyHandler.ts:34]: we should update the name of the method

File: src/event-kind-handler/PolicyHandler.ts [hunk 8]
```diff
@@ -31,7 +31,8 @@ export class PolicyHandler extends EventKindHandler {
...
+  private readonly getLabelsByPolicyId: (policy_ids: string[] | string, paginationOpts: PaginationOpts) => Promise<Map<string, PublishedTransactionMetadata | Array<PublishedTransactionMetadata>>>
```
Comment (sebastianmontero) [inline src/event-kind-handler/PolicyHandler.ts:34]: we should update the name of the method

File: src/event-kind-handler/PolicyHandler.ts [hunk 9]
```diff
@@ -31,7 +31,8 @@ export class PolicyHandler extends EventKindHandler {
...
+  private readonly saveLabel: (policyId: string, label: TransactionMetadata | Array<TransactionMetadata>) => Promise<Array<PublishedTransactionMetadata>>
```
Comment (sebastianmontero) [inline src/event-kind-handler/PolicyHandler.ts:35]: we should update the name of the method

**Review by sebastianmontero — CHANGES_REQUESTED** [review]
It looks good, but I think we should more appropriately name code that is still referencing "label" which is now transaction metadata

**Comment by Dreyhh** [comment]
@sebastianmontero the changes requested are included in the commit 254db9bc0f6fccd130fb06a66f5fddfea7c894ff

**Pull request merged**